File: iosApp/iosApp/Screens/Register/RegisterView.swift
//
//  RegisterView.swift
//  iosApp
//
//  Created by MacBook on 12.02.2024.
//  Copyright © 2024 orgName. All rights reserved.
//

import Foundation
import SwiftUI
import SharedSDK

struct RegisterView: View {
    
    let viewState:RegisterViewState
    let eventHandler: (RegisterEvent) -> Void
    
    
    @State var login = ""
    @State var password = ""
    @State var name = ""
    
    var body: some View {
        
        VStack{
            TextField("name", text: $name)
                .onChange(of: login){value in
                    eventHandler(RegisterEvent.NameChanged(value: (name)))
                }
            
            TextField("login", text: $login)
                .onChange(of: login){value in
                    eventHandler(RegisterEvent.EmailChanged(value: (login)))
                }
            TextField("password", text: $password)
                .onChange(of: password){value in
                    eventHandler(RegisterEvent.PasswordChanged(value: (password)))
                }
            
            Button(action: {
                eventHandler(RegisterEvent.RegisterClick())
            }, label: {
                Text("send data")
            })
            
            Button(action: {
                eventHandler(RegisterEvent.LoginClick())
            }, label: {
                Text("to login")
            })
        }
        
    }
}

File: common/auth/compose/src/androidMain/kotlin/register/RegisterView.kt
package register

import androidx.compose.foundation.layout.Column
import androidx.compose.material.Button
import androidx.compose.material.Text
import androidx.compose.material.TextField
import androidx.compose.runtime.Composable
import login.models.LoginEvent

import registration.models.RegisterEvent
import registration.models.RegisterViewState

@Composable
fun RegisterView(state: RegisterViewState, eventHandler: (RegisterEvent) -> Unit) {
    Column {

        TextField(
            value = state.name,
            onValueChange = {
                eventHandler(RegisterEvent.NameChanged(it))
            }
        )

        TextField(
            value = state.login,
            onValueChange = {
                eventHandler(RegisterEvent.EmailChanged(it))
            }
        )

        TextField(
            value = state.password,
            onValueChange = {
                eventHandler(RegisterEvent.PasswordChanged(it))
            }
        )

        Button(
            onClick = {
                eventHandler(RegisterEvent.RegisterClick)
            }
        ){
            Text("Click")
        }

        Button(
            onClick = {
                eventHandler(RegisterEvent.LoginClick)
            }
        ){
            Text("to login")
        }

    }
}